exporter/awsemfexporter/conn.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020, OpenTelemetry Authors
+// Portions of this file Copyright 2018-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsemfexporter
+
+import (
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"go.uber.org/zap"
+)
+
+type connAttr interface {
+	newAWSSession(logger *zap.Logger, roleArn string, region string) (*session.Session, error)
+	getEC2Region(s *session.Session) (string, error)
+}
+
+// Conn implements connAttr interface.
+type Conn struct{}
+
+func (c *Conn) getEC2Region(s *session.Session) (string, error) {
+	return ec2metadata.New(s).Region()
+}
+
+// AWS STS endpoint constants
+const (
+	STSEndpointPrefix         = "https://sts."
+	STSEndpointSuffix         = ".amazonaws.com"
+	STSAwsCnPartitionIDSuffix = ".amazonaws.com.cn" // AWS China partition.
+)
+
+// GetAWSConfigSession returns AWS config and session instances.
+func GetAWSConfigSession(logger *zap.Logger, cn connAttr, cfg *Config) (*aws.Config, *session.Session, error) {
```

Comment:
Whoever merges all the AWS conn.go files into one under `internal/aws` will get a prize